File: src/model/response.go
// @author xiangqian
// @date 20:24 2023/03/22
package model

import (
	"encoding/gob"
)

// Response 响应数据
type Response struct {
	Msg  string `json:"msg"`  // 消息（没有消息就是最好的消息）
	Data any    `json:"data"` // 数据
}

// 注册模型
func init() {
	gob.Register(Response{})
}

File: src/main/java/org/xiangqian/note/model/Response.java
package org.xiangqian.note.model;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author xiangqian
 * @date 15:07 2024/05/03
 */
@Data
@NoArgsConstructor
public class Response<T> {

    private String code;
    private String msg;
    private T data;

    private Response(String code, String msg, T data) {
        this.code = code;
        this.msg = msg;
        this.data = data;
    }

    public static <T> Response<T> ok(T data) {
        return new Response<>("ok", null, data);
    }

    public static <T> Response<T> error(String msg) {
        return new Response<>("error", msg, null);
    }

}
